src/merkle/empty_nodes.rs

```diff
@@ -0,0 +1,649 @@
+use super::{Felt, RpoDigest, WORD_SIZE, ZERO};
+use core::slice;
+
+// EMPTY NODES SUBTREES
+// ================================================================================================
+
+/// An empty sub-trees set.
+///
+/// Will provide precomputed node values for path computation for pre-defined depths.
+pub struct EmptyNodesSubtrees;
+
+impl EmptyNodesSubtrees {
+    // PROVIDERS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns a static slice that is the opening from zero until the target depth.
+    ///
```

Comment:
nit: I'd phrase it in terms of empty subtrees rather than openings. Maybe something like `Returns an static slice with roots of empty subtrees of a Merkle tree starting at the specified depth."

Reply:
Done!